time/src/test/java/net/jqwik/time/api/LocalDateTimeTests.java

```diff
@@ -0,0 +1,349 @@
+					return true;
+				});
+
+			}
+
+			@Property
+			void between(@ForAll("dateTimes") LocalDateTime min, @ForAll("dateTimes") LocalDateTime max, @ForAll Random random) {
+
+				Assume.that(!min.isAfter(max));
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().between(min, max);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isAfterOrEqualTo(min);
+					assertThat(dateTime).isBeforeOrEqualTo(max);
+					return true;
+				});
+
+			}
+
+			@Property
+			void betweenMinAfterMax(@ForAll("dateTimes") LocalDateTime min, @ForAll("dateTimes") LocalDateTime max, @ForAll Random random) {
+
+				Assume.that(min.isAfter(max));
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().between(min, max);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isAfterOrEqualTo(max);
+					assertThat(dateTime).isBeforeOrEqualTo(min);
+					return true;
+				});
+
+			}
+
+			@Property
+			void betweenSame(@ForAll("dateTimes") LocalDateTime same, @ForAll Random random) {
+
+				Arbitrary<LocalDateTime> dateTimes = DateTimes.dateTimes().between(same, same);
+
+				assertAllGenerated(dateTimes.generator(1000, true), random, dateTime -> {
+					assertThat(dateTime).isEqualTo(same);
+					return true;
+				});
+
+			}
+
+		}
+
+	}
+
+	@Group
+	class Shrinking {
+
+		@Property
+		void defaultShrinking(@ForAll Random random) {
+			LocalDateTimeArbitrary dateTimes = DateTimes.dateTimes();
+			LocalDateTime value = falsifyThenShrink(dateTimes, random);
+			assertThat(value).isEqualTo(LocalDateTime.of(1900, Month.JANUARY, 1, 0, 0, 0));
+		}
+
+		@Disabled("Not working at the moment.")
+		@Property
+		void shrinksToSmallestFailingValue(@ForAll Random random) {
+			LocalDateTimeArbitrary dateTimes = DateTimes.dateTimes();
+			TestingFalsifier<LocalDateTime> falsifier = dateTime -> dateTime.isBefore(LocalDateTime.of(2013, Month.MAY, 25, 13, 12, 55));
+			LocalDateTime value = falsifyThenShrink(dateTimes, random, falsifier);
+			assertThat(value).isEqualTo(LocalDateTime.of(2013, Month.MAY, 25, 13, 12, 55));
+		}
+
+	}
+
+	@Group
+	class ExhaustiveGeneration {
+
+		@Example
+		void between() {
+			Optional<ExhaustiveGenerator<LocalDateTime>> optionalGenerator =
+					DateTimes.dateTimes()
+							 .between(
+									 LocalDateTime.of(2013, 5, 25, 11, 22, 31, 392_211_322),
+									 LocalDateTime.of(2013, 5, 25, 11, 22, 35, 392_211_325)
+							 )
+							 .exhaustive();
+			assertThat(optionalGenerator).isPresent();
+
+			ExhaustiveGenerator<LocalDateTime> generator = optionalGenerator.get();
+			assertThat(generator.maxCount()).isEqualTo(4);
+			assertThat(generator).containsExactly(
+					LocalDateTime.of(2013, 5, 25, 11, 22, 32),
+					LocalDateTime.of(2013, 5, 25, 11, 22, 33),
+					LocalDateTime.of(2013, 5, 25, 11, 22, 34),
+					LocalDateTime.of(2013, 5, 25, 11, 22, 35)
+			);
+		}
+
+	}
+
+	@Group
+	class EdgeCasesTests {
```

Comment:
Edge cases for between(min, max) should produce min and max, but I think they don't.
Adding such a test should reveal if I'm right.

Reply:
Yes, you're right, I overlooked that.